models/typescript/v3.0/tests/v3.0.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { createCheckers } from "ts-interface-checker";
+
+// checker model
+import GbfsVersionsTI from './ti/gbfs_versions-ti';
+import GbfsTI from './ti/gbfs-ti';
+import GeofencingZonesTI from './ti/geofencing_zones-ti';
+import ManifestTI from './ti/manifest-ti';
+import StationsInformationTI from './ti/station_information-ti';
+import StationStatusTI from './ti/station_status-ti';
+import SystemAlertsTI from "./ti/system_alerts-ti";
+import SystemInformationTI from './ti/system_information-ti';
+import SystemPricingPlansTI from './ti/system_pricing_plans-ti'
+import SystemRegionsTI  from "./ti/system_regions-ti";
+import VehicleStatusTI from "./ti/vehicle_status-ti";
+import VehicleTypesTI from './ti/vehicle_types-ti';
+
+// checkers
+const { GbfsVersions } = createCheckers(GbfsVersionsTI);
+const { Gbfs } = createCheckers(GbfsTI);
+const { GeofencingZones } = createCheckers(GeofencingZonesTI);
+const { Manifest } = createCheckers(ManifestTI);
+const { StationInformation } = createCheckers(StationsInformationTI);
+const { StationStatus } = createCheckers(StationStatusTI);
+const { SystemAlerts } = createCheckers(SystemAlertsTI);
+const { SystemInformation } = createCheckers(SystemInformationTI);
+const { SystemPricingPlans } = createCheckers(SystemPricingPlansTI);
+const { SystemRegions } = createCheckers(SystemRegionsTI);
+const { VehicleStatus } = createCheckers(VehicleStatusTI);
+const { VehicleTypes } = createCheckers(VehicleTypesTI);
+
+// json test data: these are gbfs with no errors for v3.0
+import gbfsVersionsJson from './data/gbfs_versions.json';
+import gbfsJson from './data/gbfs.json';
+import geofencingZonesJson from './data/geofencing_zones.json';
+import manifestJson from './data/manifest.json';
+import stationInformationJson from './data/station_information.json';
+import stationStatusJson from './data/station_status.json';
+import systemAlertsJson from './data/system_alerts.json';
+import systemInformationJson from './data/system_information.json';
+import systemPricingPlansJson from './data/system_pricing_plans.json';
+import systemRegionsJson from './data/system_regions.json';
+import vehicleStatusJson from './data/vehicle_status.json';
+import vehicleTypesJson from './data/vehicle_types.json';
+
+
+function isValidV3Date(dateString: any): boolean {
+    if (typeof dateString !== 'string') {
+        return false;
+    }
+    const date = new Date(dateString);
+    return !isNaN(date.getTime());
+}
+
+// Date objects cannot be represented in JSON
```

Comment:
Looking back at this, it might cause an issue. The schema-to-.ts interpreter turns

gbfs.json (schema)
```
"last_updated": {
      "description": "Last time the data in the feed was updated in RFC3339 format.",
      "type": "string",
      "format": "date-time"
},
```

to
gbfs.d.ts (interface)
```
last_updated: Date;
```
but the reality is the data looks like 
gbfs.json (data)
```
"last_updated": "2024-04-18T09:37:59.000+00:00",
```

Reply:
@davidgamez do you think we should force replace all `Date` with `string` types?